bgpd/bgp_packet.c

```diff
@@ -2028,6 +2071,81 @@ static int bgp_route_refresh_receive(struct peer *peer, bgp_size_t size)
 				   SUBGRP_STATUS_DEFAULT_ORIGINATE);
 	}
 
+	/* Enhanced Route Refresh (rfc7313) */
+	if (subtype & BGP_ROUTE_REFRESH_BORR) {
+		if (peer->t_refresh_stalepath) {
+			zlog_err(
+				"%s: %d/%d, BoRR msg received, whereas BoRR already received",
+				peer->host, afi, safi);
+			return BGP_PACKET_NOOP;
+		}
+
+		/* Set routes to stale */
+		SET_FLAG(peer->sflags, PEER_STATUS_ENHANCED_REFRESH);
```

Comment:
PEER_STATUS_ENHANCED_REFRESH should be set per AFI-SAFI and it should only be set that AFI-SAFI is negotiated with this peer.